src/logging.cpp

```diff
@@ -0,0 +1,90 @@
+/**
+ * @file src/logging.cpp
+ * @brief Logging implementation file for the Sunshine application.
+ */
+
+// standard includes
+#include <iostream>
+
+// lib includes
+#include <boost/log/attributes/clock.hpp>
+#include <boost/log/common.hpp>
+#include <boost/log/expressions.hpp>
+#include <boost/log/sinks.hpp>
+#include <boost/log/sources/severity_logger.hpp>
+
+// local includes
+#include "logging.h"
+#include "version.h"
+
+using namespace std::literals;
+
+namespace help {
+  int
+  entry(const char *name, int argc, char *argv[]) {
+    print_help(name);
+    return 0;
+  }
+}  // namespace help
+
+namespace version {
+  int
+  entry(const char *name, int argc, char *argv[]) {
+    // just exit, the version has already been printed
+    return 0;
+  }
+}  // namespace version
```

Comment:
I think the help and version stuff should stay in main.cpp, since it's essentially part of the entrypoint of Sunshine.exe. For tests, you'll want your own entrypoint with its own help and other arguments, so you don't want to pull that it via `logging.cpp` when you build your tests.

Reply:
Okay. I have to separate `print_help` because it's used in config.cpp.

With the unit test PR, I need to include all of the executable sources except `main.cpp` (since it contains `main()`, but if I drop `main.cpp` then anything that relies on it will fail. The suggestion from [here](https://cuhkszlib-xiaoxing.readthedocs.io/en/latest/external/gtest/googletest/docs/FAQ.html#how-do-i-test-a-file-that-defines-main) is to have a file with nothing but `main()`.

These two namespaces don't necessarily have to be in `logging` though... I could add something like `main_helper` or `arg_processor`... What do you think of something like that? (Actually there are a couple others that could be moved as well... `restore_nvprefs_undo` and `gen_creds`)

Side note: `restore_nvprefs_undo` seems to be an undocumented argument. I should probably update the `print_help` for that.

Reply:
I moved the `help` and `version` namespace back to `main.cpp` for now. Also added a commit to update the `print_help`.

Reply:
`restore_nvprefs_undo` is an internal argument that we pass to ourselves when relaunching to restore settings. I don't think we need to document it because it's only intended for internal use by Sunshine itself.

Reply:
Alright, I removed that.